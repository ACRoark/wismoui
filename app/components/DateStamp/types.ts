export type DateStampFormat = 'auto' | 'long' | 'short';
