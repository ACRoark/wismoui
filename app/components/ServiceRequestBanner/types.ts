export interface IBannerNotification {
  hasError: boolean;
  message: IMessage;
}

export interface IMessage {
  id: string;
  values: string[] | null;
}
