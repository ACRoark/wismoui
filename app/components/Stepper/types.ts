import { ReactElement } from 'react';
import { Description } from 'types';

export interface IStepDescription {
  description: Description;
  key: string;
  status: StepperStatus;
  title: ReactElement | string;
}

export enum StepperStatus {
  Active = 'active',
  Completed = 'completed',
  Error = 'error',
  Pending = 'pending',
}
