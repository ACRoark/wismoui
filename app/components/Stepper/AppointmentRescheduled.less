@import (reference) '../../styles/themes/index.less';

@appointment-rescheduled-cls-prefix: ~'@{dte-prefix}-appointment-rescheduled';

.@{appointment-rescheduled-cls-prefix} {
  .message {
    font-weight: @dte-wismo-font-weight-semi-bold;
  }
}
