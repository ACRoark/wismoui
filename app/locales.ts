export const DEFAULT_LOCALE = 'en';

const PROD_LOCALES = {
  en: 'English',
};

const TEST_LOCALES = {
  en: 'English',
  es: 'Spanish',
};

export const SUPPORTED_LOCALES = process.env.NODE_ENV === 'production' ? PROD_LOCALES : TEST_LOCALES;
