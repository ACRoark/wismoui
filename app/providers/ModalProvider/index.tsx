/**
 *
 * ModalProvider
 *
 */
import React, { useState } from 'react';

import IModalConfig from 'types/IModalConfig';
import IModalState from 'types/IModalState';

import Modal from './Modal';

import 'react-responsive-modal/styles.css';

interface IModalProviderProps {
  children?: React.ReactNode;
}

const ModalContext = React.createContext<IModalState | null>(null);

const ModalConsumer = ModalContext.Consumer;

const ModalProvider: React.FC<IModalProviderProps> = (props: IModalProviderProps): React.ReactElement => {
  const { children } = props;

  const [modalConfig, setModalConfig] = useState<IModalConfig | null>(null);

  const showModal = (config: IModalConfig): void => {
    setModalConfig(config);
  };

  const hideModal = (): void => {
    setModalConfig(null);
  };

  return (
    <ModalContext.Provider
      value={{
        hideModal,
        showModal,
      }}
    >
      {React.Children.only(children)}
      {modalConfig && <Modal onClose={hideModal} config={modalConfig} />}
    </ModalContext.Provider>
  );
};

export { ModalConsumer, ModalContext, ModalProvider };
