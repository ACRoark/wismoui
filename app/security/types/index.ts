import IAuthenticationState from 'types/IAuthenticationState';

import IAuthenticationResult from './IAuthenticationResult';

export { IAuthenticationResult };

export default IAuthenticationState;
