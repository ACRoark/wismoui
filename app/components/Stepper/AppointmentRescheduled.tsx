import React from 'react';
import { FormattedMessage } from 'react-intl';

import DateStamp from 'components/DateStamp';
import { ProductType, SlotType } from 'types';

import Appointment from './Appointment';
import messages from './messages';

import './AppointmentRescheduled.less';

interface IAppointmentRescheduledProps {
  date: string;
  product: ProductType;
  slotType: SlotType;
}

const AppointmentRescheduled: React.FC<IAppointmentRescheduledProps> = (
  props: IAppointmentRescheduledProps,
): React.ReactElement => {
  const { date, product, slotType } = props;

  return (
    <div className="dte-wismo-appointment-rescheduled">
      <span className="message">
        <FormattedMessage
          {...messages.RESCHEDULED}
          values={{
            date: <DateStamp format="long" value={date} />,
          }}
        />
      </span>
      <Appointment key={product} slotType={slotType} product={product} />
    </div>
  );
};

export default AppointmentRescheduled;
