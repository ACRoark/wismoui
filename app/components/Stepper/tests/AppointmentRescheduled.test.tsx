import React from 'react';
import { Provider } from 'react-redux';

import { SUPPORTED_LOCALES } from 'locales';

import { ModalProvider } from 'providers/ModalProvider';
import { createMockStore, createSnapshotWithIntl } from 'testing/utils';
import { IServiceOrderEvent, ProductType, SlotType } from 'types';

import AppointmentRescheduled from '../AppointmentRescheduled';

const expectComponentToMatchSnapshot = (
  product: ProductType,
  slotType: SlotType,
  date: string,
): void => {
  const languages = Object.keys(SUPPORTED_LOCALES);
  languages.forEach((language: string): void => {
    const tree = createSnapshotWithIntl(
      <Provider store={createMockStore()}>
        <ModalProvider>
          <AppointmentRescheduled product={product} slotType={slotType} date={date} />
        </ModalProvider>
      </Provider>, language
    );
    expect(tree).toMatchSnapshot();
  });
};

describe('<AppointmentResheduled />', (): void => {

  const serviceOrderEvents: IServiceOrderEvent[] = [
    {
      appointment: {
        date: '20200622',
        slotType: 'AM',
      },
      latestStatus: {
        createdAtDate: '20200622',
        createdAtTime: '223443',
        serviceOrderEventStatus: 'RESCHEDULED',
      },
      product: 'EFR_D1',
      serviceOrderCategory: null,
      serviceOrderId: 'abc123',
    },
  ];

  it('should render correctly when the required props are given', (): void => {
    const {appointment: {slotType, date}, product} = serviceOrderEvents[0];

    expectComponentToMatchSnapshot(product, slotType, date);
  });
});
