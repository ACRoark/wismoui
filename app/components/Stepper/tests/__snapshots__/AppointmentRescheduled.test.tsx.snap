// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppointmentResheduled /> should render correctly when the required props are given 1`] = `
<div
  className="dte-wismo-appointment-rescheduled"
>
  <span
    className="message"
  >
    Rescheduled to 
    Monday
    , 
    June
     
    22
  </span>
  <div>
    Electric appointment 
    8 a.m. - 12 p.m.
  </div>
</div>
`;

exports[`<AppointmentResheduled /> should render correctly when the required props are given 2`] = `
<div
  className="dte-wismo-appointment-rescheduled"
>
  <span
    className="message"
  >
    Reprogramado para 
    Lunes
    , 
    Junio
     
    22
  </span>
  <div>
    Cita electrica 
    8 a.m. - 12 p.m.
  </div>
</div>
`;
